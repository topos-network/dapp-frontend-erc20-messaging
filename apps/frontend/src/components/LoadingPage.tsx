import { Space, Spin, Typography } from 'antd';
import React from 'react';

const { Text } = Typography;

interface Props {
  endpoint: string;
}

export default ({ endpoint }: Props) => (
  <div
    style={{
      position: 'fixed',
      top: '50%',
      left: '50%',
      transform: 'translate(-50%, -50%)',
      textAlign: 'center',
    }}
  >
    <Space direction="vertical">
      <Spin size="large" />
      <Text code>Connecting to a Topos subnet node at {endpoint}</Text>
    </Space>
  </div>
);
