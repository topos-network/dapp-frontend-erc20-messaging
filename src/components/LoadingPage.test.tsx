import React from 'react'
import { render } from '@testing-library/react'

import LoadingPage from './LoadingPage'

test('Should match snapshot', () => {
  const { asFragment } = render(<LoadingPage endpoint="" />)
  expect(asFragment()).toMatchSnapshot()
})
