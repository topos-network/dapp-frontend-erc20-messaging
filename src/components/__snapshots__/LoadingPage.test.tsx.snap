// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should match snapshot 1`] = `
<DocumentFragment>
  <div
    style="position: fixed; top: 50%; left: 50%; transform: translate(-50%, -50%); text-align: center;"
  >
    <div
      class="ant-space ant-space-vertical"
    >
      <div
        class="ant-space-item"
        style="margin-bottom: 8px;"
      >
        <div
          aria-busy="true"
          aria-live="polite"
          class="ant-spin ant-spin-lg ant-spin-spinning"
        >
          <span
            class="ant-spin-dot ant-spin-dot-spin"
          >
            <i
              class="ant-spin-dot-item"
            />
            <i
              class="ant-spin-dot-item"
            />
            <i
              class="ant-spin-dot-item"
            />
            <i
              class="ant-spin-dot-item"
            />
          </span>
        </div>
      </div>
      <div
        class="ant-space-item"
      >
        <span
          class="ant-typography"
        >
          <code>
            Connecting to a Topos subnet node at 
          </code>
        </span>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
